app/uk/gov/hmrc/helptosavefrontend/controllers/EligibilityCheckController.scala

```diff
@@ -108,28 +108,30 @@ class EligibilityCheckController @Inject() (val messagesApi:             Message
 
   private def performEligibilityChecks(nino: NINO)(implicit hc: HeaderCarrier, htsContext: HtsContext): EitherT[Future, Error, EligibilityResultWithUserInfo] =
     for {
-      userDetailsURI ← EitherT.fromOption[Future](htsContext.userDetailsURI, Error("Could not find user details URI"))
       eligible ← helpToSaveService.checkEligibility(nino).leftMap(Error.apply)
-      resultWithInfo ← getUserInformation(eligible, nino, userDetailsURI)
+      resultWithInfo ← getUserInformation(eligible, nino)
```

Comment:
we can actually get the user info before doing the eligibility check rather than the other way around - we can avoid calling the backend if we know we can't get the user information. Also, getting the user information now doesn't depend on the elilgibility result